lib/puppet/application/device.rb

```diff
@@ -198,6 +207,16 @@ def main
       if options[:target]
         devices.select! { |key, value| key == options[:target] }
       end
+      unless options[:apply].nil?
+        if options[:target].nil?
+          Puppet.err _("missing argument: --target, When using --apply")
```

Comment:
nit, `When` capitalization is a bit odd. Perhaps "`--target` is required when using `--apply`"?